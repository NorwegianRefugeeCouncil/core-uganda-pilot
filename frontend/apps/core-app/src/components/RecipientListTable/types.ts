import {
  ColumnInstance,
  TableInstance,
  TableState,
  UseGlobalFiltersInstanceProps,
  UseGlobalFiltersState,
  UseSortByColumnProps,
  UseSortByInstanceProps,
} from 'react-table';

export interface SortedFilteredTable<T extends Record<string, any>>
  extends Omit<
      TableInstance,
      'flatRows' | 'rows' | 'rowsById' | 'columns' | 'state'
    >,
    UseGlobalFiltersInstanceProps<T>,
    UseSortByInstanceProps<T>,
    UseGlobalFiltersState<T> {
  columns: Array<ColumnInstance & UseSortByColumnProps<T>>;
  state: TableState & UseGlobalFiltersState<T>;
}

export type RecipientListTableEntry = Record<string, any>;
