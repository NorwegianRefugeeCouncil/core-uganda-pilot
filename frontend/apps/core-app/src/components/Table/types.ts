import {
  ColumnInstance,
  TableInstance,
  TableState,
  UseGlobalFiltersInstanceProps,
  UseGlobalFiltersState,
  UseSortByColumnProps,
  UseSortByInstanceProps,
} from 'react-table';

export interface TableProps
  extends Omit<
      TableInstance,
      'flatRows' | 'rows' | 'rowsById' | 'columns' | 'state'
    >,
    UseGlobalFiltersInstanceProps<any>,
    UseSortByInstanceProps<any>,
    UseGlobalFiltersState<any> {
  columns: Array<ColumnInstance & UseSortByColumnProps<any>>;
  state: TableState & UseGlobalFiltersState<any>;
}
