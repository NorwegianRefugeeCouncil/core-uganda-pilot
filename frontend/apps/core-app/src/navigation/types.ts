export type RootParamList = {
  RecipientList: Record<string, never>;
  RecipientRegistration: Record<string, never>;
  RecipientProfile: { id: string };
};
