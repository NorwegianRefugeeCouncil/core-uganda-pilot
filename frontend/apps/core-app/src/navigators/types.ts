export type RootStackParamList = {
  RecipientList: Record<string, never>;
  RecipientRegistration: Record<string, never>;
  RecipientProfile: { id: string };
};
