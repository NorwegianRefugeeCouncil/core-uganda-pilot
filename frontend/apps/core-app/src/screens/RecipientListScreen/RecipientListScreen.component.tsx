import * as React from 'react';
import { Button, Text, VStack } from 'native-base';
import { NavigationProp, RouteProp } from '@react-navigation/native';

import { RootParamList } from '../../navigators/types';

import * as Styles from './RecipientListScreen.styles';

type Props = {
  route: RouteProp<RootParamList, 'RecipientList'>;
  navigation: NavigationProp<RootParamList>;
};

export const RecipientListScreenComponent: React.FC<Props> = ({
  route,
  navigation,
}) => {
  return (
    <Styles.Container>
      <Text variant="display">{route.name}</Text>
      <VStack space={2} width="sm">
        <Button
          variant="major"
          onPress={() =>
            navigation.navigate('RecipientProfile', {
              id: '1',
            })
          }
        >
          Recipient 1
        </Button>
        <Button
          variant="major"
          onPress={() =>
            navigation.navigate('RecipientProfile', {
              id: '2',
            })
          }
        >
          Recipient 2
        </Button>
      </VStack>
    </Styles.Container>
  );
};
