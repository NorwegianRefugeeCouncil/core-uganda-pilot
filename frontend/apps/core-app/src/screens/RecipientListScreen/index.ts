export { RecipientListScreenContainer as RecipientListScreen } from './RecipientListScreen.container';
