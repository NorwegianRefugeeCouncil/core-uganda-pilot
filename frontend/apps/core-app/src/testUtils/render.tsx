import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import {
  render as r,
  RenderAPI,
  RenderOptions,
} from '@testing-library/react-native';
import { render as rWeb } from '@testing-library/react';
import { NativeBaseProvider } from 'native-base';
import { theme } from 'core-design-system';

const inset = {
  frame: { x: 0, y: 0, width: 0, height: 0 },
  insets: { top: 0, left: 0, right: 0, bottom: 0 },
};

export const render = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
): RenderAPI =>
  r(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );

export const renderWeb = (
  component: React.ReactElement<any>,
  options?: RenderOptions | undefined,
) =>
  rWeb(
    <NativeBaseProvider theme={theme} initialWindowMetrics={inset}>
      <NavigationContainer>{component}</NavigationContainer>
    </NativeBaseProvider>,
    options,
  );
