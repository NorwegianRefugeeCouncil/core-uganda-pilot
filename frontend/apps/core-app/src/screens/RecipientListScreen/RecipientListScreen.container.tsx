import * as React from 'react';
import { StackScreenProps } from '@react-navigation/stack';

import { RootNavigatorParamList } from '../../navigation/root';
import { useRecipientForms } from '../../contexts/RecipientForms';

import { RecipientListScreenComponent } from './RecipientListScreen.component';

type Props = StackScreenProps<RootNavigatorParamList, 'recipientsList'>;

export const RecipientListScreenContainer: React.FC<Props> = () => {
  const recipientForms = useRecipientForms();

  return (
    <RecipientListScreenComponent
      forms={recipientForms}
      isLoading={!recipientForms || !recipientForms.length}
    />
  );
};
