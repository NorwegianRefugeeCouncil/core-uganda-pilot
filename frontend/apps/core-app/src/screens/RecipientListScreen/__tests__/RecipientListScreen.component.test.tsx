import { NavigationProp, useNavigation } from '@react-navigation/native';
import React from 'react';

import { render } from '../../../testUtils/render';
import { RecipientListScreenComponent } from '../RecipientListScreen.component';
import { RootParamList } from '../../../navigators/types';

it('should match the snapshot', () => {
  const Wrapper: React.FC = () => {
    const navigation = useNavigation<NavigationProp<RootParamList>>();
    return (
      <RecipientListScreenComponent
        route={{ name: 'RecipientList', params: {}, key: 'key' }}
        navigation={navigation}
      />
    );
  };

  const { toJSON } = render(<Wrapper />);
  expect(toJSON()).toMatchSnapshot();
});
