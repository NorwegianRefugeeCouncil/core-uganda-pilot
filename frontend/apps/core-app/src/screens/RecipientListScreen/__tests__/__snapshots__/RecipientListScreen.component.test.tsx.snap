// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match the snapshot 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
>
  <View
    dataSet={Object {}}
    style={
      Array [
        Object {},
        Array [
          Object {},
        ],
      ]
    }
  >
    <Text
      dataSet={Object {}}
      style={
        Object {
          "backgroundColor": undefined,
          "color": "#1f2937",
          "fontFamily": "Roboto_700Bold",
          "fontSize": 44,
          "fontStyle": "normal",
          "fontWeight": "400",
          "letterSpacing": 0,
          "lineHeight": 52,
          "textAlign": "auto",
          "textDecorationLine": undefined,
        }
      }
    >
      RecipientList
    </Text>
    <View
      dataSet={Object {}}
      style={
        Object {
          "width": 384,
        }
      }
    >
      <View
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FF7602",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": undefined,
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 38,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              Recipient 1
            </Text>
          </View>
        </View>
      </View>
      <View
        dataSet={Object {}}
        style={
          Object {
            "height": 10,
          }
        }
      />
      <View
        _text={
          Object {
            "color": "lightText",
          }
        }
        accessibilityRole="button"
        accessible={true}
        collapsable={false}
        colorScheme="primary"
        dataSet={Object {}}
        focusable={true}
        onBlur={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onHoverIn={[Function]}
        onHoverOut={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#FF7602",
            "borderRadius": 4,
            "flexDirection": "row",
            "justifyContent": "center",
            "paddingBottom": 10,
            "paddingLeft": 15,
            "paddingRight": 15,
            "paddingTop": 10,
          }
        }
      >
        <View
          dataSet={Object {}}
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
            }
          }
        >
          <View
            dataSet={Object {}}
            style={Object {}}
          >
            <Text
              dataSet={Object {}}
              style={
                Object {
                  "backgroundColor": undefined,
                  "color": "#FFFFFF",
                  "fontFamily": undefined,
                  "fontSize": 18,
                  "fontStyle": "normal",
                  "fontWeight": "700",
                  "letterSpacing": 0,
                  "lineHeight": 38,
                  "textAlign": "auto",
                  "textDecorationLine": undefined,
                }
              }
            >
              Recipient 2
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaProvider>
`;
